### PR TITLE
ci(commitlint): Apply feedback to commitlint github action

### PR DESCRIPTION
**Pull request by spetrosi**
- Add commitlint.config.js as a file for visibility
- Do not pull the config
- Do not set redundant env vars, just use them inline
- Quote github.event.pull_request.title in case it contains shell metacharacter
- Do not ignore tests/roles/candlepin in rhc role

### CODE REVIEW

File: playbooks/files/.github/commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+module.exports = {
```
Comment (spetrosi) [inline playbooks/files/.github/commitlint.config.js:1]: @richm is it fine to store commitlint config in `.github` dir?

Comment (richm) [reply playbooks/files/.github/commitlint.config.js:1]: > @richm is it fine to store commitlint config in `.github` dir?

I think `playbooks/files/` is more appropriate - especially if we implement this checking in `tox` or in some other git pre-commit check.

File: playbooks/templates/.github/workflows/commitlint.yml [hunk 2]
```diff
@@ -22,43 +22,26 @@ jobs:
...
+      #
+      # One cannot use the number of commits
+      # (github.event.pull_request.commits) to find the start commit
+      # i.e. HEAD~N does not work, this breaks if there are merge commits.
+      - name: Run commitlint on commits
+        run: >-
 {%- raw %}
-          echo "PR_HEAD=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
-          echo "PR_BASE=${{ github.event.pull_request.base.sha }}" >> $GITHUB_ENV
+          npx commitlint --from ${{ github.event.pull_request.base.sha }}
```
Comment (richm) [inline playbooks/templates/.github/workflows/commitlint.yml:37]:
```suggestion
          npx commitlint --from '${{ github.event.pull_request.base.sha }}'
```

File: playbooks/templates/.github/workflows/commitlint.yml [hunk 3]
```diff
@@ -22,43 +22,25 @@ jobs:
...
+      #
+      # One cannot use the number of commits
+      # (github.event.pull_request.commits) to find the start commit
+      # i.e. HEAD~N does not work, this breaks if there are merge commits.
+      - name: Run commitlint on commits
+        run: >-
 {%- raw %}
-          echo "PR_HEAD=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
-          echo "PR_BASE=${{ github.event.pull_request.base.sha }}" >> $GITHUB_ENV
+          npx commitlint '--from ${{ github.event.pull_request.base.sha }}'
```
Comment (richm) [inline playbooks/templates/.github/workflows/commitlint.yml:37]:
```suggestion
          npx commitlint --from '${{ github.event.pull_request.base.sha }}'
```

File: playbooks/templates/.github/workflows/commitlint.yml [hunk 4]
```diff
@@ -22,43 +22,25 @@ jobs:
...
+      # One cannot use the number of commits
+      # (github.event.pull_request.commits) to find the start commit
+      # i.e. HEAD~N does not work, this breaks if there are merge commits.
+      - name: Run commitlint on commits
+        run: >-
 {%- raw %}
-          echo "PR_HEAD=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
-          echo "PR_BASE=${{ github.event.pull_request.base.sha }}" >> $GITHUB_ENV
+          npx commitlint --from '${{ github.event.pull_request.base.sha }}'
+          --to '${{ github.event.pull_request.head.sha }}' --verbose
```
Comment (richm) [inline playbooks/templates/.github/workflows/commitlint.yml:38]: so `npx commitlint` will automatically look for the file `.commitlintrc.js`?

Comment (spetrosi) [reply playbooks/templates/.github/workflows/commitlint.yml:40]: yes, it's one of the supported filenames for config file

**Review by richm — APPROVED** [review]

**Pull request merged**